upsilonconf/io.py

```diff
@@ -225,26 +225,99 @@ def _get_io_function(path: Path, write: bool = False):
     raise ValueError(f"unknown config file extension: '{ext}'")
 
 
-def load(path: Union[Path, str]) -> Configuration:
+def __replace_in_keys(
+    dictionary: Union[Mapping[str, Any], list, Any], s: str, r: str
+) -> Union[Mapping[str, Any], list, Any]:
+    """
+    Take a list or dictionary and replace all occurrencies of `s` in any of its
+    keys with `r`.
+
+    This function is called recursively.
+
+    Modelled after the Stack Overflow answer by Farhan Haider:
+    https://stackoverflow.com/questions/21650850/pyyaml-replace-dash-in-keys-with-underscore#answer-55986782
+
+    Parameters
+    ----------
+    dictionary : dict|list
+        The list or dictionary to be modified.
+    s : str
+        The string to be replaced.
+    r : str
+        The replacement string.
+
+    Returns
+    -------
+        The object of the same type with all strings `s` replaced by `r`.
+    """
+    final_dict = {}
+    for key, value in dictionary.items():
+        if isinstance(dictionary[key], (dict, list)):
```

Comment:
Maybe one more thing: you will probably want to check for `Mapping` here (and drop `list`).
Alternatively, we could also use a `try` block (which would probably be considered more pythonic)

Reply:
No problem. I thought, you wanted to have all checks removed …

Solved in https://github.com/hoedt/upsilonconf/pull/3/commits/49f8604208253f8e17f6aa40741c7965a4105f56.

Reply:
I believe there was a misunderstanding:

My suggestion was to insert either

```python
if isinstance(value, Mapping):
    value = __replace_in_keys(value, s, r)
```
or

```python
try:
    value = __replace_in_keys(value, s, r)
except AttributeError:
    pass
```

inside of the loop

Reply:
My last commit was complete rubbish. That happens, if you do it quick-quick between some other tasks. So sorry!

An updated -- and hopefully correct – version should be here: https://github.com/hoedt/upsilonconf/pull/3/commits/7e6f251a8b427a6924a6d3760195f16497d7e210.

Reply:
No problem. However, I would only catch `AttributeError` and no other exceptions. After all, that should be the exception that is raised when calling `.items()`.

Reply:
Voilà! https://github.com/hoedt/upsilonconf/pull/3/commits/3e89f36c68b7605e8fc03e70c25410eebc7126ce